hashintel/hash
# Python Temporal worker with dev infrastructure

## 🌟 What is the purpose of this PR?



This PR adds a python worker for the Temporal cluster that is able to pick up tasks off the `aipy` task queue. This can be considered a scaffold/template for now, and will eventually take over the `hash-agents` app.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1204274556967336/1204521571620724/f) _(internal)_

## 🚫 Blocked by




## 🔍 What does this change?




- Adds Temporal Python worker app
- Changes dev compose to start the worker
- Adds python worker build steps to CI
- Adjusts documentation

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:













### 📜 Does this require a change to the docs?




The changes in this PR:






- [x] requires changes to docs
    - Documentation for running the python worker 

### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:







## ⚠️ Known issues



The workers operate in separate work queues so that we don't run unnecessary retries. We should in theory be able to add them to the same task queue and have them work identically as their implementation is more or less the same

## 🐾 Next steps



## 🛡 What tests cover this?



## ❓ How to test this?



1.  Checkout the branch 
1.  Ensure you have an OpenAI key set in your `.env.local`
1.  Start external services with `yarn external-services up`
1.  Using the temporal CLI, run `temporal workflow execute -t aipy --type DemoWorkflow -i '"say hi"` which should execute the workflow defined in the python runner

## 📹 Demo


![image](https://github.com/hashintel/hash/assets/6988668/723d81a6-e356-4615-a00e-25c699305ea9)
